### PR TITLE
keys: add api key type and some fixes

### PR DESCRIPTION
**Pull request by sanderpick**
- Adds a key type to API keys (account and user types)
- Fixes CLI commands that should not be picking up the thread ID from a bucket config
- Adds NPM publishing for `buckets-grpc` and `users-grpc`
- Some other minor nits

### CODE REVIEW

File: collections/sessions_test.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ func TestSessions_Touch(t *testing.T) {
 	created, err := col.Create(context.Background(), owner)
 	require.Nil(t, err)
 
+	time.Sleep(time.Second)
```
Comment (sanderpick) [inline collections/sessions_test.go:52]: Oop, waiting a bit here to stop the flakiness in this test.

File: .gitignore [hunk 2]
```diff
@@ -16,9 +16,8 @@ vendor/
 
 # JS
 node_modules/
-**/javascript/src/*
-!**/javascript/src/index.ts
-**/javascript/dist/
+**/javascript/*.js
+**/javascript/*.d.ts
```
[the remark this answers is not in the document]
Comment (carsonfarmer) [reply .gitignore:20]: Is there a .gitignore entry for them?

[the remark this answers is not in the document]
Comment (sanderpick) [reply .gitignore:20]: This is the .gitignore file ;)

[the remark this answers is not in the document]
Comment (carsonfarmer) [reply .gitignore:20]: 🤦 Wow, that's embarrassing!

[the remark this answers is not in the document]
Comment (sanderpick) [reply .gitignore:20]: Haha, nah.. it looks like the Makefile here

**Pull request merged**